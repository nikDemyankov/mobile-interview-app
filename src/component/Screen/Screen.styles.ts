import { StyleSheet } from 'react-native';
import { color } from '../../ui/theme';

const styles = StyleSheet.create({
  screen: {
    flex: 1,
    backgroundColor: color.background,
  },
  children: {
    paddingHorizontal: 10,
    paddingVertical: 10,
  },
});

export default styles;
