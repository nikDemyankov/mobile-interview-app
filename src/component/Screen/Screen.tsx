import React from 'react';
import styled from 'styled-components/native';
import { color } from '../../ui/theme';

const Container = styled.SafeAreaView({
  flex: 1,
  backgroundColor: color.background,
});

const ChildrenContainer = styled.View({
  paddingHorizontal: 10,
  paddingVertical: 10,
});

interface Props {
  children: React.ReactNode | React.ReactNode[];
}

const Screen: React.FC<Props> = ({ children }) => {
  return (
    <Container>
      <ChildrenContainer>{children}</ChildrenContainer>
    </Container>
  );
};

export default Screen;
