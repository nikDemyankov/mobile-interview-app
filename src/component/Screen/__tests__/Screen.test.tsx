import React from 'react';
import { Text, View } from 'react-native';
import { create } from 'react-test-renderer';

import Screen from '../Screen';

it('should render properly with one child', () => {
  const component = create(
    <Screen>
      <View />
    </Screen>,
  );

  expect(component).toMatchSnapshot();
});

it('should render properly with several children', () => {
  const component = create(
    <Screen>
      <View />
      <Text>'Hello'</Text>
    </Screen>,
  );

  expect(component).toMatchSnapshot();
});
