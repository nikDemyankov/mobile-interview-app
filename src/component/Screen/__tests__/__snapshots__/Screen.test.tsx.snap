// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render properly with one child 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "backgroundColor": "#FFF",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "paddingHorizontal": 10,
          "paddingVertical": 10,
        },
      ]
    }
  >
    <View />
  </View>
</RCTSafeAreaView>
`;

exports[`should render properly with several children 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "backgroundColor": "#FFF",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "paddingHorizontal": 10,
          "paddingVertical": 10,
        },
      ]
    }
  >
    <View />
    <Text>
      'Hello'
    </Text>
  </View>
</RCTSafeAreaView>
`;
